File: Greedy/1029. Two City Scheduling/1029.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        refund = []
        n = len(costs) // 2

        res = 0
        for a,b in costs:
            refund.append(b - a)
            res += a
        
        refund.sort(key=lambda x:(x))
        for i in range(n):
            res += refund[i]
        
        return res

File: Greedy/1029. Two City Scheduling/1029.cpp
class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);
        
        vector<int> res;
        int minCost = 0;
        for (int i=0; i<costs.size(); ++i) {
            res.push_back(costs[i][1] - costs[i][0]);
            minCost += costs[i][0];
        }
        sort(res.begin(),res.end());

        for (int i=0; i<costs.size()/2; ++i) {
            minCost += res[i];
        }

        return minCost;
    }
};